pkg/resources/stream_on_external_table_acceptance_test.go

```diff
@@ -414,6 +416,108 @@ func TestAcc_StreamOnExternalTable_CopyGrants(t *testing.T) {
 	})
 }
 
+func TestAcc_StreamOnExternalTable_RecreateWhenStale(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+	resourceName := "snowflake_stream_on_external_table.test"
+
+	schema, cleanupSchema := acc.TestClient().Schema.CreateSchemaWithOpts(t,
+		acc.TestClient().Ids.RandomDatabaseObjectIdentifierInDatabase(acc.TestClient().Ids.DatabaseId()),
+		&sdk.CreateSchemaOptions{
+			DataRetentionTimeInDays:    sdk.Pointer(0),
+			MaxDataExtensionTimeInDays: sdk.Pointer(0),
+		},
+	)
+	t.Cleanup(cleanupSchema)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schema.ID())
+
+	stageID := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schema.ID())
+	stageLocation := fmt.Sprintf("@%s", stageID.FullyQualifiedName())
+	_, stageCleanup := acc.TestClient().Stage.CreateStageWithURL(t, stageID)
+	t.Cleanup(stageCleanup)
+
+	externalTable, externalTableCleanup := acc.TestClient().ExternalTable.CreateInSchemaWithLocation(t, stageLocation, schema.ID())
+	t.Cleanup(externalTableCleanup)
+
+	var createdOn string
+
+	model := model.StreamOnExternalTable("test", id.DatabaseName(), externalTable.ID().FullyQualifiedName(), id.Name(), id.SchemaName()).WithInsertOnly(r.BooleanTrue)
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnExternalTable),
+		Steps: []resource.TestStep{
```

Comment:
I would like to see two different tests here:
1. The one you put here but modified slightly and informing that permadiff will always occur with such a setup.
2. The test with a bit more sophisticated setup:
- initial creation does not lead to stale stream (e.g. setting the schema params initially to non-zero value)
- changing the value externally on schema and checking manually (through client) that stream turned stale
- changing schema parameters back to non-zero values (+ optionally ensuring that the stream is still stale and did not magically unstale on its own)
- running the terraform to check that we recreate the stale stream and expecting no changes after

Reply:
1. Not sure what you mean - how modified exactly?
2. Added.

Reply:
1. FIrst of all name. And secondly, if there is a permadiff, we don't need that many steps (create and next step should be enough)

Reply:
I'm still not sure what you mean by that, but I changed this (only for this resource for now).

Reply:
One more comment on the new test (closing this thread).